### PR TITLE
Add an HTTPS agent for localhost request

### PR DESCRIPTION
**Pull request by Bill7zz**
Resolve #XXX

### Description
To allow connections from services on HTTPS localhost, we added a validation placed on top of the fetch requests that injects an HTTPS agent when the Emulator prompts a request to the localhost port. 

### Changes made

- Add https agent for localhost request
- Fix typescript/no-var-requires when require HTTPS

### Testing
image

### CODE REVIEW

**Comment by GasparAcevedoZainSouthworks** [comment]
`To allow connections from services on HTTPS localhost. we added...` -> To allow connections from services on HTTPS localhost`,` we added ...

**Comment by GasparAcevedoZainSouthworks** [comment]
Also, there is another typo:
`... an HTPPS agent when the Emulator prompt a request ...` -> _...  an HTPPS agent when the Emulator prompt_`s` _a request  ..._

**Comment by GasparAcevedoZainSouthworks** [comment]
In the **Changes made** section, can you replace the _https_ with **HTTPS**?

File: packages/app/main/src/fetchProxy.ts [hunk 1]
```diff
@@ -38,6 +38,15 @@ const nodeFetch = require('node-fetch');
 declare function fetch(input: RequestInfo, init?: RequestInit): Promise<Response>;
 (global as any).fetch = function(...args: any[]) {
   const [urlOrRequest, requestInit = {}] = args;
+
+  // Https localhost
+  // eslint-disable-next-line typescript/no-var-requires
+  const https = require('https');
+  const httpsAgent = new https.Agent({ rejectUnauthorized: false });
+  if (args[0].includes('https://localhost')) {
```
Comment (Aliandi) [inline packages/app/main/src/fetchProxy.ts]: I would try and have the string stored somewhere else in a var instead of hardcoded here.

**Comment by LCarbonellRod** [comment]
@Bill7zz in the section "**Changes Made**" i think that in  `... Fix typescript/no-var-requires when require HTTPS...`  instead of `require` is `requires`

**Comment by Aliandi** [comment]
HTPPS -> HTTPS?

File: packages/app/main/src/fetchProxy.ts [hunk 2]
```diff
@@ -38,6 +38,16 @@ const nodeFetch = require('node-fetch');
 declare function fetch(input: RequestInfo, init?: RequestInit): Promise<Response>;
 (global as any).fetch = function(...args: any[]) {
   const [urlOrRequest, requestInit = {}] = args;
+
+  // Https localhost
+  // eslint-disable-next-line typescript/no-var-requires
+  const https = require('https');
+  const httpsAgent = new https.Agent({ rejectUnauthorized: false });
+  const allowLocalhost = 'https://localhost';
+  if (args[0].includes(allowLocalhost)) {
```
Comment (denscollo) [inline packages/app/main/src/fetchProxy.ts]: there'd be a space between the const declarations and the if clause

Comment (Bill7zz) [reply packages/app/main/src/fetchProxy.ts]: Updated

**Comment by Bill7zz** [comment]
MS PR#1779 merged

**Pull request closed without merging**